src/django_components/component.py

```diff
@@ -54,37 +55,113 @@
+        # So we reused that and convert it to user-defined Media class
+        media_prop: property = cls.media
+        media_getter = media_prop.fget
+        def media_wrapper(self: "Component") -> Any:
+            if not media_getter:
+                return None
+            media: Media = media_getter(self)
+            return self.media_class(js=media._js, css=media._css)
+        cls.media = property(media_wrapper)
 
-            # Allow: class Media: js = "script.js"
-            if hasattr(media, "js") and isinstance(media.js, str):
-                media.js = [media.js]
+        return cls
 
-        _resolve_component_relative_files(attrs)
 
-        return super().__new__(mcs, name, bases, attrs)
+def _normalize_media(media: "Component.Media") -> None:
+    if hasattr(media, "css") and media.css:
+        # Allow: class Media: css = "style.css"
+        if _is_media_filepath(media.css):
+            media.css = [media.css]
+
+        # Allow: class Media: css = ["style.css"]
+        if isinstance(media.css, list):
+            media.css = {"all": media.css}
+
+        # Allow: class Media: css = {"all": "style.css"}
+        if isinstance(media.css, dict):
+            for media_type, path_list in media.css.items():
+                if _is_media_filepath(path_list):
+                    media.css[media_type] = [path_list]  # type: ignore
+
+    if hasattr(media, "js") and media.js:
+        # Allow: class Media: js = "script.js"
+        if _is_media_filepath(media.js):
+            media.js = [media.js]
+
+
+def _normalize_media_filepaths(media: "Component.Media") -> None:
+    return _map_media_filepaths(media, _normalize_media_filepath)
+
+
+def _map_media_filepaths(media: "Component.Media", map_fn: Callable[[Any], Any]) -> None:
+    if hasattr(media, "css"):
+        if not isinstance(media.css, dict):
+            raise ValueError("#TODO") # TODO
+
+        for media_type, path_list in media.css.items():
+            media.css[media_type] = list(map(map_fn, path_list))
+
+    if hasattr(media, "js"):
+        if not isinstance(media.js, list):
+            raise ValueError("#TODO") # TODO
+
+        media.js = list(map(map_fn, media.js))
+
+
+import pathlib
+import os
+from django.utils.safestring import SafeData
+def _is_media_filepath(filepath: Any) -> bool:
+    if isinstance(filepath, SafeData) or hasattr(filepath, "__html__"):
+        return True
+
+    elif isinstance(filepath, (pathlib.PurePath, os.PathLike)) or hasattr(filepath, "__fspath__"):
+        return True
+
+    if isinstance(filepath, bytes):
+        return True
+
+    if isinstance(filepath, str):
+        return True
+    
+    return False
+
+
+def _normalize_media_filepath(filepath: Any) -> str | SafeData:
+    print("_normalize_media_filepath: ", filepath)
+
+    if isinstance(filepath, SafeData) or hasattr(filepath, "__html__"):
+        return filepath
```

Comment:
This bit above ensures that when users creates an "object path" with `@ html_safe`, then we don't try to convert it to a string.